cmd/object-handlers.go

```diff
@@ -191,7 +191,7 @@ func (api objectAPIHandlers) SelectObjectContentHandler(w http.ResponseWriter, r
 		rs := &HTTPRangeSpec{
 			IsSuffixLength: isSuffixLength,
 			Start:          offset,
-			End:            offset + length,
+			End:            offset + length - 1,
```

Comment:
This is an incorrect change @mailsmail - all the tests have failed.

Reply:
@harshavardhana I've already fixed it. It fails when the length is -1, because then the `End` will be negative for offset 0.